From 9c626f97b25bf1973ece91c6f5dd67628bb0abd1 Mon Sep 17 00:00:00 2001
From: David Teigland <teigland@redhat.com>
Date: Fri, 18 Oct 2024 12:57:49 -0500
Subject: [PATCH] systemd-wdmd: work around race with udev setting softdog
 label

After modprobe softdog, udev wants to set permission and/or label
on the watchdog device.  Wait until that's done before trying to
test/probe the watchdog device.  RHEL-56479
---
 init.d/systemd-wdmd | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/init.d/systemd-wdmd b/init.d/systemd-wdmd
index 001609736287..c3126c408a89 100644
--- a/init.d/systemd-wdmd
+++ b/init.d/systemd-wdmd
@@ -19,6 +19,8 @@ watchdog_check() {
 	if [ $retval -ne 0 ]; then
 		echo -n $"Loading the softdog kernel module: "
 		modprobe softdog
+		# handle delay in udev setting permission/label
+		udevadm settle
 
 		watchdog_probe
 		retval=$?
-- 
2.46.2

